editor/editor_properties.cpp

```diff
@@ -1689,93 +1689,155 @@ EditorPropertyEasing::EditorPropertyEasing() {
 	add_child(spin);
 }
 
-///////////////////// VECTOR2 /////////////////////////
+///////////////////// VECTORN /////////////////////////
 
-void EditorPropertyVector2::_set_read_only(bool p_read_only) {
-	for (int i = 0; i < 2; i++) {
-		spin[i]->set_read_only(p_read_only);
+const String EditorPropertyVectorN::COMPONENT_LABELS[4] = { "x", "y", "z", "w" };
+
+int EditorPropertyVectorN::_get_ratio_component(int p_idx, int p_component) const {
+	int i = p_idx / (component_count - 1);
+	if (p_component == 1) {
+		return i;
+	} else {
+		return (i + p_idx % (component_count - 1) + 1) % component_count;
 	}
 }
 
-void EditorPropertyVector2::_value_changed(double val, const String &p_name) {
-	if (setting) {
-		return;
+void EditorPropertyVectorN::_set_read_only(bool p_read_only) {
+	for (EditorSpinSlider *spin : spin_sliders) {
+		spin->set_read_only(p_read_only);
 	}
+}
 
+void EditorPropertyVectorN::_value_changed(double val, const String &p_name) {
 	if (linked->is_pressed()) {
-		setting = true;
-		if (p_name == "x") {
-			spin[1]->set_value(spin[0]->get_value() * ratio_yx);
+		for (int i = 0; i < component_count; i++) {
+			if (p_name == COMPONENT_LABELS[i]) {
+				for (int j = 0; j < ratio.size(); j++) {
+					if (_get_ratio_component(j, 1) == i) {
+						for (int k = 0; k < component_count - 1; k++) {
+							spin_sliders[_get_ratio_component(j + k, 0)]->set_value_no_signal(spin_sliders[_get_ratio_component(j + k, 1)]->get_value() * ratio[j + k]);
+						}
+						break;
+					}
+				}
```

Comment:
This needs to be simplified, but it took me too long to even figure it out ;_;
Maybe later >\_>

Reply:
How about this? It's longer but easier to read (command-query separation and reduced the number of identation levels)
```suggestion

		// Find out which component has changed.
		int component_idx = -1;
		for (int i = 0; i < component_count; i++) {
			if (p_name == COMPONENT_LABELS[i]) {
				component_idx = i;
			}
		}

		int ratio_component_idx = -1;
		for (int j = 0; j < ratio.size(); j++) {
			int ratio_idx = _get_ratio_component(j, 1);
			if (ratio_idx == component_idx) {
				ratio_component_idx = j;
			}
		}

		// Update the other components based on the ratio.
		if (component_idx > 0 && ratio_component_idx > 0) {
			for (int k = 0; k < component_count - 1; k++) {
				int rc1 = _get_ratio_component(ratio_component_idx + k, 1);
				int rc2 = _get_ratio_component(ratio_component_idx + k, 0);
				double value = ratio[ratio_component_idx + k] * spin_sliders[rc1]->get_value();
				spin_sliders[rc2]->set_value_no_signal(value);
			}
		}
```